src/document/url.ml

```diff
@@ -263,6 +263,11 @@ module Anchor = struct
     let page = Path.from_identifier parent in
     Ok { page; anchor = str_name; kind }
 
+  (* This is needed to ensure that references to polymorphic constructors have
+     links that use the right suffix: those resolved references are turned into
+     _constructor_ identifiers. *)
+  let suffix_for_constructor x = x
```

Comment:
I don't understand.

Reply:
Polymorphic constructors do not have an ID. When turning a resolved polymorphic constructor reference to an ID, a normal constructor ID is used.
But then, we need to ensure that this id, when turned to an href with anchor, does correspond link to the right place (that is, the page exists (which is always the case but also the html anchor exists).
Anchors are generated in `Url.Anchor.polymorphic_variant`, more precisely [here](https://github.com/ocaml/odoc/pull/1115/files#diff-d955f7dbfb34dd3b6f73f9c626306d7240006429f385bb1de02bc55183087595R422-R424). This function just tries to "force" the target anchor in IDs for constructor and the anchor for polymorphic constructor to correspond. I can remove it and replace it by a more detailed comment.

I can also add IDs to polymorphic constructors that can be referenced, that means changes to `Lang` but maybe it is worth it.